src/Xamarin.CodeAnalysis/XI0004.cs

```diff
@@ -0,0 +1,177 @@
+				var baseType = method.ContainingType.BaseType;
+				ISymbol baseMethod = method.OverriddenMethod;
+				// returns if the [RequiresSuper] attribute isn't present
+				if (!baseMethod.GetAttributes ().Any (attr => (attr.AttributeClass.Name == RequiresSuperAttributeName && attr.AttributeClass.ContainingNamespace.Name == RequiresSuperAttributeNamespace)))
+					return;
+
+				foreach (var block in codeBlockContext.OperationBlocks.Where ((b) => b.Kind == OperationKind.Block)) // loops once per method
+				{
+					foreach (var exp in block.Children) // loops once per operation in the method body 
+					{
+						//skips if the operation isn't base.___ or return ___; 
+						if (exp.Kind != OperationKind.ExpressionStatement && exp.Kind != OperationKind.Return)
+							continue;
+
+						foreach (var child in exp.Children) {
+							// skips if it's not an Invocation Operation
+							if (child.Kind != OperationKind.Invocation) 
+								continue;
+							var invocationOp = (IInvocationOperation)child;
+							var instance = invocationOp.Instance;
+							if (instance == null) // FIXME: this is the void foo () => . . .case; make this register a diagnostic (the code fix currently throws an error if the diagnostic is registered)
+								return;
+							// skips if the operation doesn't contain base
+							if (instance.Kind != OperationKind.InstanceReference)
+								continue;
+							var instanceOp = (IInstanceReferenceOperation)instance;
+
+							// checks if in the base.[method]() call, [method] matches the one in the method's signature
+							if (instanceOp.ReferenceKind != InstanceReferenceKind.ContainingTypeInstance)
+								continue;
+							var invokedMethod = invocationOp.TargetMethod;
+							if (invokedMethod.Equals(method.OverriddenMethod))
+							{
+								return;
+							}
+						}
+					}
+				}
+				//adds the diagnostic if we don't find base.method() or return base.method()
+				var location = codeBlockContext.OwningSymbol.Locations.FirstOrDefault ();
+				if (location == null)
+					return;
+				var diagnostic = Diagnostic.Create (Rule, location, method.Name);
+				codeBlockContext.ReportDiagnostic (diagnostic);
+			});
+		}
+
+	}
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(RequiresSuperAnalyzerCodeFixProvider))]
+    [Shared]
+    public class RequiresSuperAnalyzerCodeFixProvider : CodeFixProvider
+    {
+        const string Title = "Add base method call";
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds
+        {
+            get { return ImmutableArray.Create(XI0004.DiagnosticId); }
+        }
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var diagnostic = context.Diagnostics.First();
```

Comment:
So, here if the context.Dianostics is empty, we get an exception. 
```csharp
public static TSource First<TSource>(this IEnumerable<TSource> source)
{
        TSource first = source.TryGetFirst(out bool found);
        if (!found)
        {
            throw Error.NoElements();
        }

        return first;
   }
```
```suggestion
            var diagnostic = context.Diagnostics.FirstOrDefault();
```

Reply:
`CodeFixContext.Diagnostics` is not a collection or a list, it's a struct, an `ImmutableArray<T>` which can never be null: https://github.com/dotnet/roslyn/blob/master/src/Workspaces/Core/Portable/CodeFixes/CodeFixContext.cs#L46 and https://github.com/dotnet/corefx/blob/master/src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.cs